pkg/kubelet/kubelet.go

```diff
@@ -1392,7 +1392,7 @@ func (kl *Kubelet) cleanupOrphanedPodDirs(pods []*api.Pod, runningPods []*kubeco
 			continue
 		}
 		glog.V(3).Infof("Orphaned pod %q found, removing", uid)
-		if err := os.RemoveAll(kl.getPodDir(uid)); err != nil {
+		if err := util.RemoveAllSkipMountPoints(kl.getPodDir(uid)); err != nil {
```

Comment:
Hmm... kubelet already checks whether [there exists any not-yet-cleaned-up volume in the pod directory](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet.go#L1423), and skip the pod directory removal if that's true. Correct me if I am wrong, but I think your PR handles pretty much the same thing?